flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -111,6 +111,11 @@ public int getNumBuffers() {
 	}
 
 	@Override
+	public int getNumberOfUsedBuffers() {
+		return numberOfRequestedMemorySegments - getNumberOfAvailableMemorySegments();
```

Comment:
instead of `getNumberOfAvailableMemorySegments()` we could use `availableMemorySegments.size()` so we never enter a synchronized block and affect the LocalBufferPool.